### PR TITLE
Add dangerous limit option and fix some typos.

### PR DESCRIPTION
**Pull request by Jun10ng**
main changes:  add dangerous limit option, ref to this [issue](https://github.com/mosn/holmes/issues/37).
other changes:  fix some typos and break long sentences to make docs easier to read without dragging screen side to side.

### CODE REVIEW

**Comment by Jun10ng** [comment]
@taoyuanyuan

**Comment by taoyuanyuan** [comment]
Good job!
I'm very sorry that I would have thought the `cpu_dangerous_limit` would work on all dumps(cpu, mem and gourtine).
But with your implementation, each dump has a `dangerous_limit`, so I perfer the frist.

maybe we could repalce dangerous to max, is it more clearly?
```
func WithCPUDump(min int, diff int, abs int, max int)
func WithMemDump(min int, diff int, abs int, max int)
```

**Comment by taoyuanyuan** [comment]
Is it necessary to support goroutine dangerous limit or mem dangerous limit？I think when the cpu dangerous limit is reached，and all dumps  (cpu mem and goeoutine) will be stoped.
What's your opinion？

**Comment by Jun10ng** [comment]
Oh that‘s my bad, I will modify my implementation. 
But before the modification, I have some questions to ask.
>" would work on all dumps(cpu, mem and gourtine)."
If we use the same parameter to represent dangerous/max limit, then how we distinguish "percent" and "number", you know that, goroutines is a "number" and CPU is "percent".

**Comment by taoyuanyuan** [comment]
if｀the cpu dangerous limit` is a global config and control all dump action， add a option func is better.
I prefer this.

**Comment by Jun10ng** [comment]
Oh that‘s my bad, I will modify my implementation.
But before the modification, I have some questions to ask.
>"func WithMemDump(min int, diff int, abs int, max int)"

Does this  `max` parameter contained in the memory option function mean cpu_dangerous_limit`?  This function signature is a bit ambiguous. Could I move to a new option function like `WithCPUMax`

>"Is it necessary to support goroutine dangerous limit or mem dangerous limit？"

You are right, cuz if memory or goroutine leak, "Fast fail" is the right thing to do.

**Comment by Jun10ng** [comment]
> if｀the cpu dangerous limit` is a global config and control all dump action， add a option func is better. I prefer this.
ok 
lol we send msg at the same time.

**Comment by taoyuanyuan** [comment]
Maybe we can add judge on func `startDumpLoop`, no need to modify `goroutineCheckAndDump` and `memCheckAndDump`.
https://github.com/mosn/holmes/blob/798377c6d12006a4d936dc0b4a1d8291cac8dc23/holmes.go#L154

**Comment by Jun10ng** [comment]
> Maybe we can add judge on func `startDumpLoop`, no need to modify `goroutineCheckAndDump` and `memCheckAndDump`.
> 
> https://github.com/mosn/holmes/blob/798377c6d12006a4d936dc0b4a1d8291cac8dc23/holmes.go#L154

I'm sorry that I have not read the context about the function I modified.

File: holmes.go [hunk 1]
```diff
@@ -152,6 +152,11 @@ func (h *Holmes) startDumpLoop() {
 			continue
 		}
 
+		if h.opts.ReachCPUMax(cpu) {
+			h.logf("[Holmes] current cpu percent [%v] is greater than the cpu max [%v], don't create profile", cpu, h.opts.CPUMaxPercent)
+			return
```
Comment (taoyuanyuan) [inline holmes.go:157]: `return` should be `continue`, maybe next loop will less than the cpu max.

Comment (Jun10ng) [reply holmes.go]: thanks

File: holmes.go [hunk 2]
```diff
@@ -152,6 +152,11 @@ func (h *Holmes) startDumpLoop() {
 			continue
 		}
 
+		if h.opts.ReachCPUMax(cpu) {
```
Comment (taoyuanyuan) [inline holmes.go:155]: I think we can take a common name, because probably ReachCPUMax just one of the reasons to diasble dump pofile.

Comment (taoyuanyuan) [reply holmes.go:155]: h.Enable?

Comment (Jun10ng) [reply holmes.go:155]: I plan to change its name to `isAbnomal`, what's your opinion?

File: options.go [hunk 3]
```diff
@@ -29,12 +29,25 @@ type options struct {
...
+	// move may result of the system crash.
+	CPUMaxPercent int
+
 	GrOpts     *grOptions
 	MemOpts    *memOptions
 	CPUOpts    *cpuOptions
 	ThreadOpts *threadOptions
 }
 
+func (ops *options) ReachCPUMax(curCPU int) bool {
```
Comment (taoyuanyuan) [inline options.go:43]: code logic move to  h.EnableDump ?

File: holmes.go (hunk 2 above)
Comment (taoyuanyuan) [reply holmes.go:155]: isEnableDump?

Comment (Jun10ng) [reply holmes.go]: I have changed its name to EnableDump and it returns an error with specific info.

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Comment by taoyuanyuan** [comment]
Thank you for your contribution!  Good night~

**Pull request merged**